### PR TITLE
[Feature] Support use column name alias in subquery

### PR DESCRIPTION
**Pull request by HangyuanLiu**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [ ] I have checked the version labels which the pr will be auto backported to target branch
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3
  - [ ] 2.2

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 63 / 64 (98.44%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|15|16|93.75%|[499]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|19|19|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/Relation.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToStringBuilder.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/QueryRelation.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/RelationFields.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CTERelation.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ValuesRelation.java|1|1|100.00%|[]|

File: fe/fe-core/src/main/java/com/starrocks/sql/ast/ValuesRelation.java [hunk 1]
```diff
@@ -33,7 +32,7 @@ This can share the same logic as select sum(1) from table
 
     public ValuesRelation(List<ArrayList<Expr>> rows, List<String> columnOutputNames) {
         this.rows = new ArrayList<>(rows);
-        this.columnOutputNames = columnOutputNames;
+        this.setColumnOutputNames(columnOutputNames);
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/ast/ValuesRelation.java:35]: `this.explicitColumnNames = columnOutputNames; ` ?

File: fe/fe-core/src/test/java/com/starrocks/connector/analyzer/SimpleQueryAnalyzer.java [hunk 2]
```diff
@@ -333,28 +333,30 @@ public Void visitTableFunction(TableFunctionRelation node, Void scope) {
...
+                    node.setColumnOutputNames(new ArrayList<>(tableFunction.getDefaultColumnNames()));
                 }
             } else {
-                if (node.getColumnNames().size() != tableFunction.getTableFnReturnTypes().size()) {
+                if (node.getColumnOutputNames().size() != tableFunction.getTableFnReturnTypes().size()) {
                     throw new SemanticException("table %s has %s columns available but %s columns specified",
-                            node.getAlias().getTbl(), node.getColumnNames().size(), tableFunction.getTableFnReturnTypes().size());
+                            node.getAlias().getTbl(),
+                            node.getColumnOutputNames().size(),
+                            tableFunction.getTableFnReturnTypes().size());
```
Comment (packy92) [inline fe/fe-core/src/test/java/com/starrocks/connector/analyzer/SimpleQueryAnalyzer.java:353]: available columns is `tableFunction.getTableFnReturnTypes()` and specified columns is `node.getColumnOutputNames()` ?

**Review by packy92 — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 81 / 83 (97.59%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/analyzer/QueryAnalyzer.java|17|18|94.44%|[499]|
|:large_blue_circle:|com/starrocks/sql/parser/AstBuilder.java|34|35|97.14%|[5688]|
|:large_blue_circle:|com/starrocks/sql/optimizer/transformer/RelationTransformer.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/Relation.java|4|4|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/AstToStringBuilder.java|17|17|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/QueryRelation.java|3|3|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/analyzer/RelationFields.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/CTERelation.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/ast/ValuesRelation.java|2|2|100.00%|[]|

**Review by packy92 — APPROVED** [review]

**Comment by liuyehcf** [comment]
Could give an example of what you've done?

**Review by liuyehcf — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by HangyuanLiu** [comment]
> Could give an example of what you've done?

A complete example is provided in AnalyzeSingleTest, which mainly supports the definition of column aliases, such as select * from (select * from test.t0) as t(a,b,c), and the subquery column names can be changed from (v1,v2, v3) changed to (a,b,c)

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**